Add yast2_clone_system to validate generated profile on stagingY

b10n1k opened the pull request:
I updated the yaml scheduler.
I havent done any other change in test_suite.
REPO_SLE_MODULE_DEVELOPMENT_TOOLS is set to factory repo.
After installation of jing is removed.
Also I removed the xmllint.

- Related ticket: https://progress.opensuse.org/issues/49415
- Needles: N/A
- Verification run: 
http://aquarius.suse.cz/tests/1233
http://aquarius.suse.cz/tests/1231
updated: http://aquarius.suse.cz/tests/1318
nonstaging: http://aquarius.suse.cz/tests/1335#step/yast2_clone_system/31

http://aquarius.suse.cz/tests/1347 using http://download.opensuse.org/tumbleweed/repo/oss/

final VR with factory repo: http://aquarius.suse.cz/tests/1356

jknphy commented:
Could you set to WIP and retry with newer build, also using workers from O3 instead of local to see if there is any difference.

rwx788 commented:
Ok, I also remember now, why we haven't enabled this as development tools module is not build for staging, hence, we cannot install packages from it. And all of your failures are related to that.
But we also don't care from where jing or xmllint are coming from, so we can use other repos. As an alternative we can even validate xml on the worker side, which might require some magic to get tooling for the validation on them.

b10n1k commented:
> Could you set to WIP and retry with newer build? 108.1 should be fine for trying out (the isos I mean, because the whole job group seems re-triggered showing partial results, but doesn't matter if the mediums have not changed).

i used latest build iso. 108.1 is the last builld but on stagingY is  BuildY.101.1. What i try to say is that it doesnt make sense to try with 108.1

b10n1k commented:
> I would also add this module to the test suite we run for daily builds, so we could easily bisect failures if something worked in staging and didn't work in the daily build. WDYT?

i would say that it is not necessary because it is covered from https://openqa.suse.de/tests/overview?arch=&machine=&modules=yast2_clone_system&distri=sle&version=15-SP2&build=120.1&groupid=129#.

rwx788 commented:
> > I would also add this module to the test suite we run for daily builds, so we could easily bisect failures if something worked in staging and didn't work in the daily build. WDYT?
> 
> i would say that it is not necessary because it is covered from https://openqa.suse.de/tests/overview?arch=&machine=&modules=yast2_clone_system&distri=sle&version=15-SP2&build=120.1&groupid=129#.

True, whereas these setups are different we still get profile with same sections and it's even better to have more keys/properties. Therefore, I guess we can skip this part.

b10n1k commented:
> Ok, I also remember now, why we haven't enabled this as development tools module is not build for staging, hence, we cannot install packages from it. And all of your failures are related to that.
> But we also don't care from where jing or xmllint are coming from, so we can use other repos. As an alternative we can even validate xml on the worker side, which might require some magic to get tooling for the validation on them.

I would like to discuss this further as i have some grasps to get the whole picture

jknphy commented on tests/console/yast2_clone_system.pm:
```diff
@@ -68,6 +64,15 @@ sub run {
+    my $deprpm  = 'https://rpmfind.net/linux/opensuse/distribution/leap/15.1/repo/oss/noarch/relaxngDatatype-2011.1-lp151.2.1.noarch.rpm';
+
+    zypper_call "--no-gpg-checks install $deprpm";
+    #script_run "wget " . $rpmlink . $packagename . " -O " . $rpmname, output=>"get jing from $rpmling";
```
this comment?

b10n1k replied:
a9bf782d9

jknphy commented on tests/console/yast2_clone_system.pm:
```diff
@@ -42,24 +42,20 @@ sub run {
         }
-        else {
-            die "autoinst.xml does not validate for unknown reason";
-        }
```
this is intended not die here?

b10n1k replied:
yes. this is not needed as the process will fail and report the problem in a previous step and this will never be reached anyway. check http://aquarius.suse.cz/tests/1231

b10n1k commented:
@jknphy i did the modifications for the jing installation(not submitted yet, will do it soon though) but unfortunately i cant run VR due to  https://bugzilla.suse.com/show_bug.cgi?id=1160909

b10n1k commented:
@jknphy RFR without VR

jknphy commented on tests/console/yast2_clone_system.pm:
```diff
@@ -42,24 +42,20 @@ sub run {
     upload_asset $ay_profile_path;
 
-    unless (is_opensuse) {
+    unless (is_opensuse || is_staging) {
```
this block has no purpose now as you always use jing from Leap not from these repos. @rwx788 WDYT? Should we still use sle repos for sle?

rwx788 replied:
We could apply this workaround solution only in case package is not available in the system. In general I'm not a fan of hardcoded urls. And if we use one, I would prefer more static ones and not the one which will stop working once there is a new version available.
So TBH, I would simply add Factory repo, install jing from it and remove it after the test, not to pollute the setup.

rwx788 replied:
There is a problem to use `CURRENT` e.g. `ftp://openqa.suse.de/SLE-12-SP5-WE-POOL-x86_64-CURRENT-Media1/`, that got broken recently and seems there is no patch expected.

b10n1k replied:
i cant see development-tools with current anyway(ftp://openqa.suse.de/). I have tried to use another repo but I still have the gpg verification problem. if someone show me i can try again because i miss something talking about Factory

b10n1k replied:
So i tried a few times the approach with http://download.opensuse.org/tumbleweed/repo/oss/ to find out that the jing is not provided neither there. So i think we are kinda stuck with this 'ungly' solution for now I have addressed the nice catch from @jknphy and provide VR for those cases. RFR

b10n1k commented:
Actually there is in http://download.opensuse.org/tumbleweed/repo/oss/noarch but installing the repo with **zypper -n -G ar http://download.opensuse.org/tumbleweed/repo/oss/noarch dev** it didnt work

b10n1k commented:
notready

b10n1k commented:
RFR

jknphy commented on tests/console/yast2_clone_system.pm:
```diff
@@ -44,22 +44,19 @@ sub run {
         my $devel_repo = get_required_var(is_sle('>=15') ? get_repo_var_name("MODULE_DEVELOPMENT_TOOLS") : 'REPO_SLE_SDK');
-        zypper_call "ar -c $utils::OPENQA_FTP_URL/" . $devel_repo . " devel-repo";
+        my $uri        = (is_staging) ? $devel_repo : "$utils::OPENQA_FTP_URL/" . $devel_repo;
+        zypper_call "ar -c -G $uri devel-repo";
```
instead of `-G` should be the same than below, right? `--gpg-auto-import-keys`

jknphy commented on tests/console/yast2_clone_system.pm:
```diff
@@ -44,22 +44,19 @@ sub run {
+        my $uri        = (is_staging) ? $devel_repo : "$utils::OPENQA_FTP_URL/" . $devel_repo;
+        zypper_call "ar -c -G $uri devel-repo";
     }
+    zypper_call '--gpg-auto-import-keys ref' unless (is_staging);
```
when there is staging (sle/openSUSE) there is no ref of the repo, Downloading metadata should be done always or I missing something?

jknphy commented on tests/console/yast2_clone_system.pm:
```diff
@@ -44,22 +44,19 @@ sub run {
 
     unless (is_opensuse) {
         my $devel_repo = get_required_var(is_sle('>=15') ? get_repo_var_name("MODULE_DEVELOPMENT_TOOLS") : 'REPO_SLE_SDK');
```
why did you decide to hide factory repo in this variable? shouldn't be tricky now to setup in openQA? just asking, I'm not sure, hardcoding it maybe more clear.

jknphy replied:
@b10n1k  Please, do not resolve conversations that are not addressed somehow.
@rwx788 do you know what is the preference to override settings when they are coming from the repo synchronization and where this setting could be located?

jknphy replied:
Checked with @rwx788 that we should not try to override variable here, as when the build is triggered it will be overriden, so we need new variable, for example `JING_REPO` and we would be able to apply it better where we need it.

b10n1k commented:
@jknphy 4ccb59130 addresses the comments about the gpg above. I removed -G and the condition in zypper_call '--gpg-auto-import-keys ref'; and VR was successful
http://aquarius.suse.cz/tests/1360#

jknphy commented:
> @jknphy [4ccb591](https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/4ccb591306febf6e7dd69e643f68ab33e5db1e1c) addresses the comments about the gpg above. I removed -G and the condition in zypper_call '--gpg-auto-import-keys ref'; and VR was successful
> http://aquarius.suse.cz/tests/1360#

Please, next time address comment individually as we agreed. And leave a while conversations unresolved so next person or the same person reviewing can have better look. Unless it is very trivial, I would recommend that you don't close yourself specially it is needed several rounds of review. It is more practical to use this feature of resolve conversation in a latter stage or review, so we can use it to discard stuff already accomplished.

rwx788 commented on tests/console/yast2_clone_system.pm:
```diff
@@ -44,22 +44,19 @@ sub run {
+    my $output_jing = script_output "jing $xml_schema_path $ay_profile_path";
 
-    if (($rc_jing) || ($rc_xmllint)) {
+    if ($output_jing) {
```
Hmm, I'm not sure if I fully get it. Documentation states:
```
Executing script inside SUT with bash -eox (in case of serial console with bash -eo) and directs stdout (not' stderr!) to the serial console and returns the output if' the script exits with 0. Otherwise the test is set to failed. 
```

So for me it sound like you cannot do soft-failure here in case jing fails. Nevertheless, there is `proceed_on_failure` flag which we can reuse. TBH, I don't understand why you want to replace `script_run` with `script_output`.

jknphy commented:
LGTM

rwx788 commented on tests/console/yast2_clone_system.pm:
```diff
@@ -44,21 +44,21 @@ sub run {
     unless (is_opensuse) {
         my $devel_repo = get_required_var(is_sle('>=15') ? get_repo_var_name("MODULE_DEVELOPMENT_TOOLS") : 'REPO_SLE_SDK');
-        zypper_call "ar -c $utils::OPENQA_FTP_URL/" . $devel_repo . " devel-repo";
+        my $uri        = (is_staging) ? "http://download.opensuse.org/tumbleweed/repo/oss/" : "$utils::OPENQA_FTP_URL/" . $devel_repo;
```
Could you please put the comment explaining what we do here.
